### ISSUE TITLE
support async rendering

### ISSUE THREAD

**Issue by graingert** [jstransformers/jstransformer-nunjucks#35]
https://mozilla.github.io/nunjucks/api.html#asynchronous-support

**Comment by ForbesLindesay** [comment]
Looks good. Would you be up for creating a pull request for this?

**Comment by graingert** [comment]
Seems quite messy to do as the API for, `compile -> renderAsync` doesn't
exist

**Comment by ForbesLindesay** [comment]
You could implement the `renderAsync` method: https://github.com/jstransformers/jstransformer#renderasync

**Comment by graingert** [comment]
That's not allowed as a result of compile

**Comment by ForbesLindesay** [comment]
No, I mean you can have a single function `renderAsync` that does both compile and render in one go. e.g. https://github.com/jstransformers/jstransformer-rollup only implements `renderFileAsync`.

**Comment by graingert** [comment]
Hmm, this removes any benefit that users might get from precompiling their
templates though

**Comment by gijswijs** [comment]
I think I have a solution for this, but it snowballed into having to make another PR for test-jstranformer. https://github.com/jstransformers/test-jstransformer/pull/50

I implemented compileAsync. This is enough (I think) to offer the asynchronous support that nunjucks offers. At least it works in my project. But it doesn't pass the tests. This is (again, I think) because the test is wrong. The test assumes that template.fn(locals) returns a string. Hence the test
fails when that functions returns a promise, because template.fn(locals).trim() returns an error. By
wrapping it in a Promise.resolve() both the situations where template.fn(locals) returns a string or a Promise are covered. This is also how the jstranformer library works, so the tests now better mimick the behavior of the actual library.  The test-jstransformer PR also adds a switch for async testing.

If that PR gets accepted, we can update to the latest test-jstransformer release in this repo, and then this PR #42 should pass testing.
